src/modules/mavlink/mavlink_main.cpp

```diff
@@ -1898,6 +1895,15 @@ Mavlink::task_main(int argc, char *argv[])
 		case 'd':
 			_device_name = myoptarg;
 			set_protocol(SERIAL);
+
+			/* USB serial is indicated by /dev/ttyACM0*/
+			if (strcmp(_device_name, "/dev/ttyACM0") == OK || strcmp(_device_name, "/dev/ttyACM1") == OK) {
```

Comment:
Won't this be problematic for anyone actually trying to set different values for the mavlink usb instance? They'll be dependent on specifying the device before other arguments.

Reply:
@dagar 
yeah it will, you will have to specify the different values after the device.
i intentionally didn't changed the defaults [value](https://github.com/PX4/Firmware/blob/dea5afd327ccfa747bf198da2360182b9131d809/src/modules/mavlink/mavlink_main.cpp#L1840) because there's a reason why they were set for the first place, e.g. [here](https://github.com/PX4/Firmware/blob/dea5afd327ccfa747bf198da2360182b9131d809/src/modules/mavlink/mavlink_main.cpp#L2074) and they serve non usb devices as defaults.
we want different defaults just for usb. 
as i'm looking at it we a 3 options :
1. leave as is.
2. after getopt function to do checks to determine if new values have been specified or use default ones, e.g. if datarate is 0 and device is usb.
3. understand that maybe we try to exclude the mavlink over usb case, something that maybe we shouldn't do. as i think about it further more you always have to specify the different values for mavlink on the start of new mavlink stream, why is the usb case should be any different? if we do it for all other mavlink instances why not just for the usb?

another thing, on options 1 and 2 ftp will always be enabled on usb, if we don't want it to always be enabled option 3 is better because there's no default for ftp to be on and it's the users choice.

Reply:
I'd say give 2 a try, and if it makes a bigger mess we go with it as is (option 1.) I'm mainly trying to anticipate the case where someone inevitably wants USB with a different mode and wastes a day debugging.

Reply:
@dagar 
pushed the latest changes (squashed so there would be only one commit).
as you can see, i changed the usb check to use variable instead of mode. i've noticed that there seem to be a bit of confusion of what to use to determine if a connection is usb or not, device type (ACM) or mode (config), all that while there is a variable exactly for that (_is_usb_uart). 
further more, https://github.com/PX4/Firmware/pull/12428 fix won't work if mode is not config.